File: backend/crates/akashic-extraction/tests/fixtures/ruby/01-class.rb
# A simple calculator class demonstrating method extraction.
class Calculator
  def initialize(initial)
    @state = initial
  end

  def add(a, b)
    a + b
  end

  def helper
    add(1, 2)
  end

  def self.version
    "1.0"
  end
end

File: backend/crates/akashic-extraction/tests/fixtures/php/01-class.php
<?php

namespace App\Services {
    /**
     * A simple calculator class demonstrating method extraction.
     */
    class Calculator
    {
        private int $state;

        public function __construct(int $initial)
        {
            $this->state = $initial;
        }

        public function add(int $a, int $b): int
        {
            $result = $a + $b;
            return $result;
        }

        private function helper(): int
        {
            return $this->add(1, 2);
        }

        public static function version(): string
        {
            return self::release();
        }

        protected static function release(): string
        {
            return "1.0";
        }
    }
}
